test/integration/features/addons/addon-registry-route.feature

```diff
@@ -18,40 +18,56 @@ so that user can able to push their container images directly to registry and de
+     Then stdout of host shell should contain "Login Succeeded"
+
+  Scenario: User can push pulled application image to OpenShift integrated docker registry
+    Given executing "docker pull bitnami/nginx" in host shell succeeds
+      And executing "docker tag bitnami/nginx docker-registry-default.$(minishift_ip).nip.io/myproject/nginx" in host shell succeeds
+     When executing "docker push docker-registry-default.$(minishift_ip).nip.io/myproject/nginx" in host shell succeeds
+     Then executing "oc get is" in host shell succeeds
       And stdout of host shell contains "nginx"
-      And executing "oc new-app --image-stream=nginx --name=nginx" in host shell succeeds
-      And stdout of host shell contains "Success"
+
+  Scenario: User can deploy application from integrated docker registry
+     When executing "oc new-app --image-stream=nginx --name=nginx" in host shell succeeds
       And executing "oc expose svc nginx" in host shell succeeds
-     Then executing "oc set probe dc/nginx --readiness --get-url=http://:8080" in host shell succeeds
+      And executing "oc set probe dc/nginx --readiness --get-url=http://:8080" in host shell succeeds
       And service "nginx" rollout successfully within "1200" seconds
-      And with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "/" of service "nginx" in namespace "myproject" is equal to "200"
+     Then with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "/" of service "nginx" in namespace "myproject" is equal to "200"
       And with up to "10" retries with wait period of "500ms" the "body" of HTTP request to "/" of service "nginx" in namespace "myproject" contains "Welcome to nginx!"
 
-  Scenario: As a user, I can login to OpenShift integrated docker registry after restart minishift
+  Scenario: User restarts Minishift
     Given Minishift has state "Running"
      When executing "minishift stop" succeeds
      Then executing "minishift start" succeeds
-      And Minishift should have state "Running"
-      And executing "minishift addon apply registry-route" succeeds
+
+  Scenario: User can login to OpenShift integrated docker registry after restart minishift
+    Given Minishift has state "Running"
+     When executing "minishift addon apply registry-route" succeeds
      Then with up to "60" retries with wait period of "3" seconds container name "registry_docker-registry" should be "running"
-      And with up to "60" retries with wait period of "1" second command "curl -ik https://docker-registry-default.$(minishift ip).nip.io" output should contain "200 OK"
-      And executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
+      And with up to "60" retries with wait period of "1000ms" the "status code" of HTTP request to "/" of service "docker-registry" in namespace "default" is equal to "200"
```

Comment:
so many wait states in our tests

Reply:
@gbraad yeah, it is unfortunate and ugly :(